### PR TITLE
fix: Do not preselect the first radio without checked attr

### PR DESCRIPTION
**Pull request by isprogfun**
According to MDN having none radios checked in the beginning is a valid situation which is not the case for our elements https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio#Value. I've added negative index to handle this. It'll work with keyboard navigation as well. First `toggle` won't do anything without having a radioElement and it's the case with the negative index.

### CODE REVIEW

**Comment by esmailpour-hosein** [comment]
Thanks for contribution @isprogfun 
I think we should not make it the default behaviour but an option. Both because it's not the most used case and also I don't want to cause breaking change just yet. So please add another property for it to disable this default behaviour.
We also need @mihneaz approval before be able to merge this.

**Comment by mihneaz** [comment]
I hope I understood this right: we are automatically checking the first radio input in any group and we'd like to disable that functionality when needed. 
If that's the case, yes, we'd like to have that functionality but not as the default behaviour.

**Comment by isprogfun** [comment]
> So please add another property for it to disable this default behaviour.

Ok. What name would you suggest for the property then? How should it behave when you have this property and also have a radio with the `checked` parameter? What would be the priority then?

**Comment by esmailpour-hosein** [comment]
My suggestion is `disable-default-initial-select`, know it's ugly but can't think of any more descriptive name, if you or @aleks-elkin have a better suggestion, go for it 👍 
It should respect the `checked` property of the radio item, otherwise developers would have to remove this if they show the form in edit mode.

**Review by esmailpour-hosein — CHANGES_REQUESTED** [review]
My suggestion is `disable-default-initial-select`, know it's ugly but can't think of any more descriptive name, if you or @aleks-elkin have a better suggestion, go for it 👍
It should respect the `checked` property of the radio item, otherwise developers would have to remove this if they show the form in edit mode.

**Comment by isprogfun** [comment]
Hi, in the new design there is no need to have this component so let's not introduce new options and keep the existing behaviour

**Pull request closed without merging**
